### PR TITLE
Enforce Line Endings in `.editorconfig`

### PR DESCRIPTION
**Pull request by cschuchardt88**
Enforces `LF` line ending in the repo; based on `.editorconfig`.

So we stop accidentally committing `CRLF` when we edit or create file outside of `vs2022`.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#
+# Merging from the command prompt will add diff markers to the files if there
+# are conflicts (Merging from VS is not affected by the settings below, in VS
+# the diff markers are never inserted). Diff markers may cause the following 
+# file extensions to fail to load in VS. An alternative would be to treat
+# these files as binary and thus will always conflict and require user
+# intervention with every merge. To do so, just uncomment the entries below
+###############################################################################
+*.sln text eol=crlf
+*.csproj text eol=crlf
```
Comment (shargon) [inline .gitattributes:26]: We can also use LF here.

Comment (cschuchardt88) [reply .gitattributes:26]: `dotnet` requires `CRLF` or else `dotnet` cli will error and crash, I think its a bug in the cli.

Comment (shargon) [reply .gitattributes:26]: csproj works, for sln I'm not sure

**Comment by cschuchardt88** [comment]
@superboyiii @shargon @vncoelho
You guys should be run:
`git config --global core.autocrlf false`

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**